homeassistant/components/notify/instapush.py

```diff
@@ -0,0 +1,141 @@
+"""
+homeassistant.components.notify.instapush
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Instapush notification service.
+
+Configuration:
+
+To use the Instapush notifier you will need to add something like the following
+to your config/configuration.yaml
+
+notify:
+  platform: instapush
+  api_key: YOUR_APP_KEY
+  app_secret: YOUR_APP_SECRET
+  event: YOUR_EVENT
+  tracker: YOUR_TRACKER
+
+VARIABLES:
+
+api_key
+*Required
+To retrieve this value log into your account at https://instapush.im and go
+to 'APPS'.
+
+app_secret
+*Required
+To get this value log into your account at https://instapush.im and go to
+'APPS'. The 'Application ID' can be found under 'Basic Info'. Make sure that
+you have at least one event for your app.
+
+event
+*Required
+To retrieve this value log into your account at https://instapush.im and go
+to 'APPS'.
+
+tracker
+*Required
+To retrieve this value log into your account at https://instapush.im and go
+to 'APPS'.
+
+Example usage of Instapush if you have an event 'notification' and a tracker
+'home-assistant'.
+
+curl -X POST \
+    -H "x-instapush-appid: YOUR_APP_KEY" \
+    -H "x-instapush-appsecret: YOUR_APP_SECRET" \
+    -H "Content-Type: application/json" \
+      -d '{"event":"notification","trackers":{"home-assistant":"Switch 1"}}' \
+    https://api.instapush.im/v1/post
+
+Details for the API : https://instapush.im/developer/rest
+
+"""
+import logging
+import json
+
+from homeassistant.helpers import validate_config
+from homeassistant.components.notify import (
+    DOMAIN, ATTR_TITLE, BaseNotificationService)
+from homeassistant.const import CONF_API_KEY
+
+_LOGGER = logging.getLogger(__name__)
+_RESOURCE = 'https://api.instapush.im/v1/post'
+
+
+def get_service(hass, config):
+    """ Get the instapush notification service. """
+
+    if not validate_config(config,
+                           {DOMAIN: [CONF_API_KEY, 'app_secret', \
+                                'event', 'tracker']},
+                           _LOGGER):
+        return None
+
+    try:
+        import requests
+
+    except ImportError:
+        _LOGGER.exception(
+            "Unable to import requests. "
+            "Did you maybe not install the 'Requests' package?")
+
+        return None
+
+    # pylint: disable=unused-variable
+    try:
```

Comment:
Would it be possible to add credential validation to this request? That way we can notify the user on boot that something is wrong.